### PR TITLE
Code cleaning

### PR DESCRIPTION
**Pull request by Brett208**
Since I was spending some time in OP2Helper, I wanted to clean / modernize some of the functions.

@DanRStevens, if any of it seems controversial, we can just exclude it. 

I mostly stayed out of BaseBuilderV2.

-Brett

### CODE REVIEW

File: OP2Helper.cpp [hunk 1]
```diff
@@ -54,15 +54,21 @@ void CreateTubeOrWallLine(int x1, int y1, int x2, int y2, map_id type)
...
 
 	// Make sure (x1 <= x2) and (y1 <= y2)
-	if (x1 > x2){ x1 ^= x2; x2 ^= x1; x1 ^= x2;	}
-	if (y1 > y2){ y1 ^= y2; y2 ^= y1; y1 ^= y2;	}
+	if (x1 > x2) { 
+		x1 ^= x2; x2 ^= x1; x1 ^= x2;
+	}
+	if (y1 > y2) { 
+		y1 ^= y2; y2 ^= y1; y1 ^= y2;
+	}
```
Comment (DanRStevens) [inline OP2Helper.cpp:59]: Ahh yes, I was looking at this a few days ago and thought this code really needed to go. It's using the "clever" XOR swap algorithm, which is really not very readable, and not a good idea to use inline in code like this. It should be replaced by something like `std::swap(x1, x2)` and `std::swap(y1, y2)`. See [`std::swap`](https://en.cppreference.com/w/cpp/algorithm/swap). It seems there is a choice of header includes for `swap`.

File: BaseBuilder.cpp [hunk 2]
```diff
@@ -83,14 +83,13 @@ void CreateBeacons(int numBeacons, const BeaconInfo beacon[])
...
-	int numUnits = TethysGame::InitialUnits();		// Number of combat units to start with
-	map_id weaponType = (Player[playerNum].IsEden() == 0) ? mapMicrowave : mapLaser;	// Set weapon based on colony type
+	// Set weapon based on colony type
+	const map_id weaponType = (Player[playerNum].IsEden() == 0) ? mapMicrowave : mapLaser;
 
-	for (; numUnits > 0; --numUnits)
+	Unit unit;
+	for (int i = TethysGame::InitialUnits(); i > 0; --i)
 	{
-		Unit unit;
```
Comment (DanRStevens) [inline BaseBuilder.cpp]: Why was `Unit` moved outside the for loop here? This looks like an accidental move.

**Review by DanRStevens — APPROVED** [review]
Thank you for the updates. Looks like good changes. Marked two places for possible further improvement.

----

We may want to consider just scrapping BaseBuilderV2 at some point. I don't think anyone is really using it, and it was never really completed. It was supposed to offer more readable syntax, though I'm not sure if it's worth it.

File: OP2Helper.cpp (hunk 1 above)
Comment (Brett208) [reply OP2Helper.cpp:59]: But how am I supposed to practice bit twiddling now???

File: BaseBuilder.cpp [hunk 3]
```diff
@@ -86,9 +86,9 @@ void CreateInitialCombatUnits(int playerNum, int x, int y)
 	// Set weapon based on colony type
 	const map_id weaponType = (Player[playerNum].IsEden() == 0) ? mapMicrowave : mapLaser;
 
-	Unit unit;
 	for (int i = TethysGame::InitialUnits(); i > 0; --i)
 	{
+		Unit unit;
```
Comment (Brett208) [inline BaseBuilder.cpp:91]: I was waffling about where this should be. Happy either way.

**Review by Brett208 — COMMENTED** [review]
I was thinking maybe we could move BaseBuilderV2 to a branch since it isn't fully implemented yet. I guess if the branch languishes long enough it could be deleted eventually.

Thanks for implementing swap and carrying the cleaning a bit further.

**Comment by DanRStevens** [comment]
I think you're right about moving the BaseBuilderV2 code to a branch.

**Pull request merged**